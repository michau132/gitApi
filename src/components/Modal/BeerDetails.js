import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Image from '../Image';

const mediaTablet = ({ theme }) => theme.media.tablet;

const BeerDetailsCnt = styled.div`
  ${mediaTablet} {
    display: grid;
    grid-template-columns: 20% 80%;
    overflow: hidden;
  }
`;

const BeerDetailsStyle = styled.div`
  ${mediaTablet} {
    margin-left: 20px;
  }
`;

const ImgWrapper = styled.div`
  height: 220px;
  margin-bottom: 10px;
  
  ${mediaTablet} {
    margin-bottom: 0;
    height: 70%;
  }
`;

const Name = styled.h2`
  font-size: 24px;
  font-weight: bold;
  margin: 0;
`;

const Tagline = styled.p`
  font-weight: lighter;
  margin-bottom: 30px;
  position: relative;
  
  :before {
    content: '';
    display: block;
    position: absolute;
    bottom: -10px;
    left: 0;
    height: 6px;
    width: 65px;
    background: ${({ theme }) => theme.color.title}
  }
`;

const BeerAttributes = styled.div`
  margin-bottom: 10px;
  
  ${mediaTablet} {
    display: flex;
  }
`;

const BeerAttribute = styled.h4`
  font-weight: bold;
  
  ${mediaTablet} {
    margin-right: 20px;
  }
  
  span {
    font-weight: normal;
  }
`;

const Description = styled.p`
  margin-bottom: 10px;
`;


const BeerDetails = ({
  image_url, name, tagline, ibu, abv, ebc, description,
}) => (
  <BeerDetailsCnt>
    <ImgWrapper>
      <Image src={image_url} alt={name} />
    </ImgWrapper>
    <BeerDetailsStyle>
      <Name>{name}</Name>
      <Tagline>{tagline}</Tagline>
      <BeerAttributes>
        <BeerAttribute>
                IBU:
          {' '}
          <span>{ibu || 0}</span>
        </BeerAttribute>
        <BeerAttribute>
                ABV:
          {' '}
          <span>{abv || 0}</span>
        </BeerAttribute>
        <BeerAttribute>
                EBC:
          {' '}
          <span>{ebc || 0}</span>
        </BeerAttribute>
      </BeerAttributes>
      <Description>{description}</Description>
    </BeerDetailsStyle>
  </BeerDetailsCnt>

);

BeerDetails.defaultProps = {
  ibu: 0,
  abv: 0,
  ebc: 0,
};


BeerDetails.propTypes = {
  image_url: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  tagline: PropTypes.string.isRequired,
  ibu: PropTypes.number,
  abv: PropTypes.number,
  ebc: PropTypes.number,
  description: PropTypes.string.isRequired,
};


export default BeerDetails;
