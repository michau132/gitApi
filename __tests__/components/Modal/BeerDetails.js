import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import BeerDetails from '../../../src/components/Modal/BeerDetails';

describe('<BeerDetails /> component', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      id: 1,
      image_url: 'https://via.placeholder.com/150',
      name: 'Beer',
      tagline: 'Drink or die',
      ibu: 12,
      abv: 6,
      ebc: 32,
      description: 'Some description',
    };
    wrapper = shallow(<BeerDetails {...props} />);
  });

  test('renders without crashing', () => {
    expect(wrapper).toBeDefined();
  });

  test('matching snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
