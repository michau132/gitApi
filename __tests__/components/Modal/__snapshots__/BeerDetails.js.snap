// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BeerDetails /> component matching snapshot 1`] = `
<styled.div>
  <styled.div>
    <Image
      alt="Beer"
      src="https://via.placeholder.com/150"
    />
  </styled.div>
  <styled.div>
    <styled.h2>
      Beer
    </styled.h2>
    <styled.p>
      Drink or die
    </styled.p>
    <styled.div>
      <styled.h4>
        IBU:
         
        <span>
          12
        </span>
      </styled.h4>
      <styled.h4>
        ABV:
         
        <span>
          6
        </span>
      </styled.h4>
      <styled.h4>
        EBC:
         
        <span>
          32
        </span>
      </styled.h4>
    </styled.div>
    <styled.p>
      Some description
    </styled.p>
  </styled.div>
</styled.div>
`;
